src/components/TeacherComponents/LessonShareCardGroup/LessonShareCardGroup.test.tsx

```diff
@@ -0,0 +1,104 @@
+import { screen } from "@testing-library/react";
+import { useForm } from "react-hook-form";
+import userEvent from "@testing-library/user-event";
+
+import { ResourceFormProps } from "../types/downloadAndShare.types";
+
+import LessonShareCardGroup from "./LessonShareCardGroup";
+
+import { LessonShareData } from "@/node-lib/curriculum-api";
+import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";
+
+const ComponentWrapper = (props: {
+  shareableResources: LessonShareData["shareableResources"];
+  shareLink: string;
+}) => {
+  const { control, trigger } = useForm<ResourceFormProps>();
+
+  return (
+    <LessonShareCardGroup
+      {...props}
+      control={control}
+      triggerForm={trigger}
+      shareableResources={props.shareableResources}
+      shareLink={props.shareLink}
+    />
+  );
+};
+
+describe("lesson share card group", () => {
+  it("should render", () => {
+    renderWithTheme(
+      <ComponentWrapper shareableResources={[]} shareLink="www.fake.com" />,
+    );
+
+    const previewButton = screen.getByText("Preview as a pupil");
```

Comment:
Recommend using `data-testid` and `getByTestId` instead of text

Reply:
oh interesting, how come? I try and avoid using data-testids unless totally necessary because they're not 'testing as a user would', and you end up adding otherwise unnecessary props to components that often don't get cleaned up when tests change etc!